src/VectorizeLoops.cpp

```diff
@@ -896,10 +898,19 @@ class VectorSubs : public IRMutator {
                     Stmt without_likelies =
                         IfThenElse::make(unwrap_tags(op->condition),
                                          op->then_case, op->else_case);
+
+                    // scalarize() will put back all vectorized loops around the statement as serial,
+                    // but it still may happen that there are vectorized loops inside of the statement
+                    // itself which we may want to handle. All the context is invalid though, so
+                    // we just start anew for this specific statement.
+                    // NOTE(vksnk): is it possible that we want to scalarize only some of the vectorized loops
```

Comment:
Isn't this note exactly the issue this fixes? Is this comment stale?

Reply:
I've extended the comment to give a bit more details, but basically this change will try to vectorize loops which are inside of given IfThenElse(likely()) node. All of the vectorized loops outside of it will be scalarized, but I am not sure if it's really necessary, so maybe we can only scalarize some of these loops.

Reply:
Yeah, I understand, I just don't understand the difference between the NOTE and the comment above? It seems like it's talking about the same thing. I am wondering if we should just delete the NOTE and keep the comment above?

Reply:
Ah, I see. Yeah, I've removed it now.